Source/WebInspectorUI/UserInterface/Protocol/Legacy/iOS/18.4/InspectorBackendCommands.js

```diff
@@ -0,0 +1,589 @@
+InspectorBackend.registerCommand("DOM.getMediaStats", null, [{"name": "nodeId", "type": "number"}], ["mediaStats"]);
+InspectorBackend.registerEvent("DOM.documentUpdated", null, []);
+InspectorBackend.registerEvent("DOM.inspect", null, ["nodeId"]);
+InspectorBackend.registerEvent("DOM.setChildNodes", null, ["parentId", "nodes"]);
+InspectorBackend.registerEvent("DOM.attributeModified", null, ["nodeId", "name", "value"]);
+InspectorBackend.registerEvent("DOM.attributeRemoved", null, ["nodeId", "name"]);
+InspectorBackend.registerEvent("DOM.inlineStyleInvalidated", null, ["nodeIds"]);
+InspectorBackend.registerEvent("DOM.characterDataModified", null, ["nodeId", "characterData"]);
+InspectorBackend.registerEvent("DOM.childNodeCountUpdated", null, ["nodeId", "childNodeCount"]);
+InspectorBackend.registerEvent("DOM.childNodeInserted", null, ["parentNodeId", "previousNodeId", "node"]);
+InspectorBackend.registerEvent("DOM.childNodeRemoved", null, ["parentNodeId", "nodeId"]);
+InspectorBackend.registerEvent("DOM.willDestroyDOMNode", null, ["nodeId"]);
+InspectorBackend.registerEvent("DOM.shadowRootPushed", ["page"], ["hostId", "root"]);
+InspectorBackend.registerEvent("DOM.shadowRootPopped", ["page"], ["hostId", "rootId"]);
+InspectorBackend.registerEvent("DOM.customElementStateChanged", ["page"], ["nodeId", "customElementState"]);
+InspectorBackend.registerEvent("DOM.pseudoElementAdded", ["page"], ["parentId", "pseudoElement"]);
+InspectorBackend.registerEvent("DOM.pseudoElementRemoved", ["page"], ["parentId", "pseudoElementId"]);
+InspectorBackend.registerEvent("DOM.didAddEventListener", null, ["nodeId"]);
+InspectorBackend.registerEvent("DOM.willRemoveEventListener", null, ["nodeId"]);
+InspectorBackend.registerEvent("DOM.didFireEvent", ["page"], ["nodeId", "eventName", "timestamp", "data"]);
+InspectorBackend.registerEvent("DOM.powerEfficientPlaybackStateChanged", ["page"], ["nodeId", "timestamp", "isPowerEfficient"]);
+InspectorBackend.registerDOMDispatcher = InspectorBackend.registerDispatcher.bind(InspectorBackend, "DOM");
+InspectorBackend.activateDomain("DOM", ["itml", "page", "web-page"]);
+
+// DOMDebugger
+InspectorBackend.registerDomain("DOMDebugger", ["page", "worker"]);
+InspectorBackend.registerEnum("DOMDebugger.DOMBreakpointType", {SubtreeModified: "subtree-modified", AttributeModified: "attribute-modified", NodeRemoved: "node-removed"});
+InspectorBackend.registerEnum("DOMDebugger.EventBreakpointType", {AnimationFrame: "animation-frame", Interval: "interval", Listener: "listener", Timeout: "timeout"});
+InspectorBackend.registerCommand("DOMDebugger.setDOMBreakpoint", null, [{"name": "nodeId", "type": "number"}, {"name": "type", "type": "string"}, {"name": "options", "type": "object", "optional": true}], []);
+InspectorBackend.registerCommand("DOMDebugger.removeDOMBreakpoint", null, [{"name": "nodeId", "type": "number"}, {"name": "type", "type": "string"}], []);
```

Comment:
these should both be limited to `["page"]` in both the `.js` and `.json` as of [290245@main](https://commits.webkit.org/290245@main)

Reply:
That commit didn't make it into iOS 18.4 / macOS 15.4. It's correct that it's not reflected in the snapshot.

Reply:
From https://github.com/WebKit/WebKit/pull/39675
>Drive-by: mark `DOMDebugger.setDOMBreakpoint` and `DOMDebugger.removeDOMBreakpoint` as being only supported by `"page"` targets since that's how the code behaves (i.e. it'll throw an error if dispatched to a `Worker`).

Does this mean that setting a breakpoint in a Worker when inspecting an older WebKit build would error out?

Reply:
Resurrecting the original comment:

https://commits.webkit.org/290245@main didn't make it into iOS 18.4 / macOS 15.4 so its changes shouldn't be included in the snapshot.

Reply:
Acknowledged. That patch (https://commits.webkit.org/290245@main) didn't land in time for iOS 18.4. We'll make a note to merge back the protocol snapshot rewrites when making the snapshot for the release where it's included.